src/services/index.ts

```diff
@@ -0,0 +1,448 @@
+  voidTransaction,
+  updateLastSyncedEvent as dbUpdateLastSyncedEvent,
+  getLastSyncedEvent,
+  getTxOutputsFromTx,
+  markUtxosAsVoided,
+  cleanupVoidedTx,
+} from '../db';
+import getConfig from '../config';
+import logger from '../logger';
+
+export const metadataDiff = async (_context: Context, event: Event) => {
+  const mysql = await getDbConnection();
+
+  try {
+    const fullNodeEvent = event.event as FullNodeEvent;
+    const {
+      hash,
+      metadata: { voided_by, first_block },
+    } = fullNodeEvent.event.data;
+    const dbTx: DbTransaction | null = await getTransactionById(mysql, hash);
+
+    if (!dbTx) {
+      if (voided_by.length > 0) {
+        // No need to add voided transactions
+        return {
+          type: 'IGNORE',
+          originalEvent: event,
+        };
+      }
+
+      return {
+        type: 'TX_NEW',
+        originalEvent: event,
+      };
+    }
+
+    // Tx is voided
+    if (voided_by.length > 0) {
+      // Was it voided on the database?
+      if (!dbTx.voided) {
+        return {
+          type: 'TX_VOIDED',
+          originalEvent: event,
+        };
+      }
+
+      return {
+        type: 'IGNORE',
+        originalEvent: event,
+      };
+    }
+
+    // Tx was voided in the database but is not anymore
+    if (dbTx.voided && voided_by.length <= 0) {
+      return {
+        type: 'TX_UNVOIDED',
+        originalEvent: event,
+      };
+    }
+
+    if (first_block
+       && first_block.length
+       && first_block.length > 0) {
+      if (!dbTx.height) {
+        return {
+          type: 'TX_FIRST_BLOCK',
+          originalEvent: event,
+        };
+      }
+
+      return {
+        type: 'IGNORE',
+        originalEvent: event,
+      };
+    }
+
+    return {
+      type: 'IGNORE',
+      originalEvent: event,
+    };
+  } catch (e) {
+    logger.error('e', e);
+    return Promise.reject(e);
+  } finally {
+    mysql.destroy();
+  }
+};
+
+export const isBlock = (version: number): boolean => version === hathorLib.constants.BLOCK_VERSION
+      || version === hathorLib.constants.MERGED_MINED_BLOCK_VERSION;
+
+export const handleVertexAccepted = async (context: Context, _event: Event) => {
+  const mysql = await getDbConnection();
+  await mysql.beginTransaction();
+
+  try {
+    const fullNodeEvent = context.event as FullNodeEvent;
+    const now = getUnixTimestamp();
+    const { BLOCK_REWARD_LOCK } = getConfig();
+    const blockRewardLock = BLOCK_REWARD_LOCK;
+
+    // @ts-ignore
+    const {
+      hash,
+      metadata,
+      timestamp,
+      version,
+      weight,
+      outputs,
+      inputs,
+      tokens,
+      token_name,
+      token_symbol,
+    } = fullNodeEvent.event.data;
+
+    const dbTx: DbTransaction | null = await getTransactionById(mysql, hash);
+
+    // Db is already on the database, ignore...
+    if (dbTx) {
+      logger.debug(`Transaction ${hash} already on the database, ignoring..`);
+      mysql.destroy();
+
+      return;
+    }
```

Comment:
If we call `handleVertexAccepted` with an existing tx, shouldn't it be an error?

Reply:
That's a good point, it should

Added a throw in 73c8c91941e33c6c8ad8119426be7af7751c7c31